todo-frontend-angular/src/app/todo/add-todo-modal/add-todo-modal.component.ts

```diff
@@ -0,0 +1,55 @@
+import { Component, EventEmitter, Output, } from '@angular/core';
+import { FormControl,Validators, FormGroup } from '@angular/forms';
+import { NewTodoDTO } from 'src/app/models/new-todo-DTO.model';
+
+
+
+@Component({
+  selector: 'app-add-todo-modal',
+  templateUrl: './add-todo-modal.component.html',
+  styleUrls: ['./add-todo-modal.component.scss']
+})
+export class AddTodoModalComponent {
+  isVisible = false;
+  isOkLoading = false;
+  validateForm: FormGroup<{
+    title: FormControl<string | null>;
+    datePicker: FormControl<Date | null>;
+}> = new FormGroup({
+  title: new FormControl<string | null>(null, [Validators.required, Validators.minLength(1)]),
+  datePicker: new FormControl<Date | null>(null)
+}) 
+constructor(){}
+
+  @Output()
+  showModal(): void {
+    this.isVisible = true;
+  }
+  @Output()
+  Submit=new EventEmitter<NewTodoDTO>();
+  handleSubmit(): void {
+    if(this.validateForm.valid)
+    {
+      this.isOkLoading = true;
+      const newTodo: NewTodoDTO={
+        Title: this.validateForm.value.title?? "",
+        Deadline: this.validateForm.value.datePicker?.toISOString()?? "",
+        Position:0,
+        TodoListId: 8
+      }
+      this.Submit.emit(newTodo)
+      this.validateForm.reset()
+      setTimeout(() => {
```

Comment:
Oof! No setTimeOuts :D Why do you want to wait?

Reply:
For giggles, I'll remove it ;)